src/main/java/org/team1540/robot2024/commands/shooter/CounterShufflePrepare.java

```diff
@@ -0,0 +1,46 @@
+package org.team1540.robot2024.commands.shooter;
+
+import com.pathplanner.lib.util.GeometryUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ProxyCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import org.team1540.robot2024.Constants;
+import org.team1540.robot2024.subsystems.drive.Drivetrain;
+import org.team1540.robot2024.subsystems.shooter.Shooter;
+import org.team1540.robot2024.util.shooter.ShooterSetpoint;
+import org.team1540.robot2024.util.vision.AprilTagsCrescendo;
+import org.team1540.robot2024.util.vision.FlipUtil;
+
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+
+public class CounterShufflePrepare extends SequentialCommandGroup {
+    public CounterShufflePrepare(Drivetrain drivetrain, Shooter shooter) {
+        this(drivetrain::getPose, shooter);
+    }
+
+    public CounterShufflePrepare(Supplier<Pose2d> positionSupplier, Shooter shooter, double leftSetpoint, double rightSetpoint) {
+        addCommands(
+                new PrepareShooterCommand(shooter, () -> new ShooterSetpoint(
+                        Rotation2d.fromRadians(
+                                Math.atan2(Constants.Targeting.STAGE_MAX_HEIGHT + 1.5 - Constants.Shooter.Pivot.PIVOT_HEIGHT,
+                                        positionSupplier.get().getTranslation().getDistance(
+                                        AprilTagsCrescendo.getInstance().getTag(AprilTagsCrescendo.Tags.CLIMB_FAR, DriverStation.getAlliance().orElse(null) == DriverStation.Alliance.Red ? DriverStation.Alliance.Blue : DriverStation.Alliance.Red
+                                                ).toPose2d().getTranslation().plus(new Translation2d(FlipUtil.flipIfRed(2.277226), 0))
```

Comment:
what's the `2.277226` for?

Reply:
Great question, it's some weird thing for moving a pose somewhere (I told simon to put the pose in constants but he wouldn't do it so we get this instead)